### PR TITLE
add OV GPU pipeline to BuildKite CI

### PR DESCRIPTION
**Pull request by bani-intelaipg**

### CODE REVIEW

**Comment by adk9** [comment]
@googlebot I consent.

File: test/ci/buildkite/igpu.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      queue: ${BUILDKITE_AGENT_META_DATA_QUEUE}
+      name: ${BUILDKITE_AGENT_META_DATA_NAME}
+
+  - wait
+  - command: |
+      rm -rf /localdisk/buildkite/artifacts/$BUILDKITE_BUILD_ID
+    label: ":wastebasket: Cleanup"
+    agents:
+      queue: ${BUILDKITE_AGENT_META_DATA_QUEUE}
+      name: ${BUILDKITE_AGENT_META_DATA_NAME}
```
Comment (adk9) [inline test/ci/buildkite/igpu.yml:89]: Is this file needed if all the steps are now enabled? Please use cpu.yml with the plugin appropriately set.

File: test/ci/buildkite/run_pipeline.sh [hunk 2]
```diff
@@ -22,6 +22,10 @@ elif [ "${BUILDKITE_PIPELINE_NAME}" == "cpu-intel-tf" ]; then
    export TF_LOCATION=/localdisk/buildkite-agent/prebuilt_intel_tensorflow
    export NGRAPH_TF_BACKEND=CPU
    PIPELINE_STEPS+=" ${SCRIPT_DIR}/cpu.yml "
+elif [ "${BUILDKITE_PIPELINE_NAME}" == "igpu" ]; then
+   export NGRAPH_TF_BACKEND=GPU
+   export TF_LOCATION=/localdisk/buildkite-agent/prebuilt_tensorflow_2_2_0
```
Comment (adk9) [inline test/ci/buildkite/run_pipeline.sh:27]: If `TF_LOCATION` is the same for all pipelines, can you please hoist it out of the conditional?

**Review by adk9 — CHANGES_REQUESTED** [review]
Please rename the pipeline to gpu instead of igpu. Our convention is to use the OV plugin name for the pipeline.

Secondly, please add a pipeline to BK so that it shows up for this PR.

File: test/ci/buildkite/run_pipeline.sh (hunk 2 above)
Comment (kanvi-nervana) [reply test/ci/buildkite/run_pipeline.sh:27]: It differs in case of Intel-TF.

File: test/python/tests_linux_gpu.txt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# This manifest is consumed by nGraph-Bridge PyTest framework via .../test/python/conftest.py
+
+# Read comments in tests_common.txt for allowed regex/patterns
+
+###################################################
+[IMPORT]
+# Read in one/more external manifest file(s)
+# Path specified is relative to this file's path
+
+tests_linux_cpu.txt
```
Comment (kanvi-nervana) [inline test/python/tests_linux_gpu.txt:12]: Shouldn't the import be for the common.txt?

File: test/tests_linux_gpu.txt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+#  Follows --gtest_filter patterns:
+#    * --> all 
+#    MathOps.* --> all tests under MathOps class
+
+###################################################
+[IMPORT]
+# Read in one/more external manifest file(s)
+# Path specified is relative to this file's path
+
+tests_linux_cpu.txt
```
Comment (kanvi-nervana) [inline test/tests_linux_gpu.txt:14]: Same here

File: test/python/tensorflow/tests_linux_gpu.txt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+#    math_*_test --> module
+#    math_*_*_test --> module
+#    math*_test --> module
+
+###################################################
+[IMPORT]
+# Read in one/more external manifest file(s)
+# Path specified is relative to this file's path
+
+tests_linux_cpu.txt
```
Comment (kanvi-nervana) [inline test/python/tensorflow/tests_linux_gpu.txt:23]: Shouldn't the import be for the common.txt?

File: test/ci/buildkite/igpu.yml (hunk 1 above)
Comment (kanvi-nervana) [reply test/ci/buildkite/igpu.yml:89]: If using cpu.yml for all pipelines then please change the name accordingly.

**Review by kanvi-nervana — CHANGES_REQUESTED** [review]

File: test/ci/buildkite/run_pipeline.sh (hunk 2 above)
Comment (bani-intelaipg) [reply test/ci/buildkite/run_pipeline.sh]: Hoisted all but intel-tf

File: test/python/tests_linux_gpu.txt (hunk 3 above)
Comment (bani-intelaipg) [reply test/python/tests_linux_gpu.txt:12]: Reusing rules as in CPU, and then adjust the extra skips

Comment (adk9) [reply test/python/tests_linux_gpu.txt:12]: No, why? The same inclusions/exclusions as for CPU apply in the case of GPU.

File: test/ci/buildkite/run_pipeline.sh (hunk 2 above)
Comment (adk9) [reply test/ci/buildkite/run_pipeline.sh]: Then it should only be set for the Intel TF case.

File: test/python/tests_linux_gpu.txt (hunk 3 above)
Comment (adk9) [reply test/python/tests_linux_gpu.txt:12]: Actually I agree with @kanvi-nervana. In both of these cases there are too many exclusions that it may be better to base this off of `common.txt`.

Comment (kanvi-nervana) [reply test/python/tests_linux_gpu.txt:12]: If you import CPU.txt then it's exclusions will also be carried forward, right? Those exclusions might not be needed for GPU and you'll then explicitly put them in the RUN list which is not required.

Comment (bani-intelaipg) [reply test/python/tests_linux_gpu.txt:12]: Correct. Can we enable more tests as part of a separate PR? As of this PR, we will have a baseline of equivalent to CPU minus few.

File: test/ci/buildkite/cpu.yml [hunk 6]
```diff
@@ -13,6 +13,8 @@
 
   - wait
   - command: |
+      source /localdisk/buildkite/artifacts/$BUILDKITE_BUILD_ID/venv/bin/activate 
+      LD_LIBRARY_PATH=${TF_LOCATION}/artifacts/tensorflow:/localdisk/buildkite/artifacts/$BUILDKITE_BUILD_ID/lib \
```
Comment (adk9) [inline test/ci/buildkite/cpu.yml:17]: This shouldn't be needed.

File: test/python/tests_linux_gpu.txt (hunk 3 above)
Comment (kanvi-nervana) [reply test/python/tests_linux_gpu.txt:12]: It's up to you, but we are in agreement that the base for every pipeline should be common.txt

Comment (adk9) [reply test/python/tests_linux_gpu.txt:12]: More tests can be enabled as a part of a separate PR (since it's a new change) but if we agree that GPU tests shouldn't depend on the CPU tests then that change should be made as a part of this PR.

I understand that the CPU pipeline is the most permissive but at the risk of being slightly repetitive, I think it's generally a better idea to keep them separate from the outset?

File: test/ci/buildkite/cpu.yml (hunk 6 above)
Comment (adk9) [reply test/ci/buildkite/cpu.yml:17]: Will be removed in a follow-up PR.

File: test/python/tensorflow/tests_linux_gpu.txt (hunk 5 above)
Comment (adk9) [reply test/python/tensorflow/tests_linux_gpu.txt:23]: Will be fixed in a follow-up PR.

File: test/tests_linux_gpu.txt (hunk 4 above)
Comment (adk9) [reply test/tests_linux_gpu.txt:14]: Will be fixed in a follow-up PR.

**Review by adk9 — APPROVED** [review]
